PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,119 @@
+
+There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions.
+
+**Please *read* every word on *every* bullet point before checking off the corresponding box.**
+
+## Minimum Prerequisites
+- [ ] The board passes ERC and DRC.
+    - Note some ERC warnings are acceptable (**_CAUTION_** please be careful)
+    - Note some DRC warnings are acceptable (less so than ERC: **_CAUTION_** please be careful).
+- [ ] All traces are routed.
+- [ ] Units are in metric.
+- [ ] Gerbers are in a zip file in the main directory (after physical design is done)
+
+**Please *read* every word on *every* bullet point before checking off the corresponding box.**
+
+## Schematic Level Requirements 
+- [ ] Is proper noise resistance given to all peripheral devices (bypass caps and coils/ferrites)?  
+- [ ] Is proper ESD protection given to all MCU input pins (zener diodes)?  
+- [ ] Is proper power protection given to peripheral devices (zener diodes)?  
+- [ ] Are peripheral units used properly (reading datasheet)?  
+- [ ] Are testing points added at useful places?  
+- [ ] Is there proper short to GND protection at MCU outputs (inline resistors)?  
+- [ ] Do ADC inputs have caps?  
+- [ ] Are ADC inputs biased so there is room above expected value to determine if value is being overflowed?  
+- [ ] Are LED's located at useful places (comm, power, debugging, extra GPIO)?  
+- [ ] Are parts chosen easy to collect?  
+- [ ] Are parts chosen easy to solder?  
+- [ ] Is there reverse polarity protection on inputs?  
+- [ ] Do parts in the schematic match the actual part datasheet?
+
+**Please *read* every word on *every* bullet point before checking off the corresponding box.**
+
+## BOM Requirements
+- [ ] **Bill Of Materials**: is there an Excel Sheet BOM and a Mouser Cart link?
+- [ ] Are *all* the correct parts in the BOM?
+- [ ] Is each part in the Mouser Cart in stock (or do we have the part in stock)?
+- [ ] Does each part in the Mouser Cart have a margin for part quantity?
+- [ ] Does every component in the schematic have a footprint, mouser link/datasheet, and reference number associated with it?
+
+**Please *read* every word on *every* bullet point before checking off the corresponding box.**
+
+## Layout Level Requirements 
+### 2D Spacing
+- [ ] The components are spaced out at an optimal distance.
+    - All components can be easily hand-soldered.
+    - IPC-SM-782A Standard requires a minimum distance of 1.0mm from edge to edge.
+- [ ] Components that are in parallel with each other are spaced out at an equal distance when possible.
+- [ ] The components are aligned with each other when possible.
+- [ ] Components are grouped based off of functionality.
+    - E.g. all components for CAN should be grouped.
+- [ ] Bypass capacitors are less than 1cm away from their respective IC's power pins.
```

Comment:
Something to check for is perhaps if the bypass/decoupling caps are:
- sized for the expected noise on the line
- sized for the expected voltage and temperature derating (particularly important for MLCCs, and HV systems)
- if multiple decoupling caps in parallel, smallest value is closest to the pin

Reply:
tbh, not sure how to check for that stuff. I normally just choose values that the datasheet recommends or just standard values.

Reply:
the first is pretty difficult without measuring a board already in prod and being in the "Real World"(TM), but the second generally has datasheets that show capacitance derating w/ voltage and temperature: particularly MLCCs, which means that if your device runs outside of standard conditions the performance and potentially more important, lifetime of the capacitors degrade - a lot

https://nepp.nasa.gov/files/25843/2013-Teverovsky-pres-MLCCsBME-n263.pdf
http://www.interfacebus.com/How_to_Derate_Capacitor.html

I don't know enough about the reliability engineering to have a good scientific method for choosing capacitors and chips to run for at least X years, but sometimes the thought of capacitors failing short or w.e. and then blowing up boards in a rocket or solar car keep me up at night.